packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -54,6 +57,73 @@ List<String> debugDescribeTransform(Matrix4 transform) {
   return matrix;
 }
 
+bool _extensionsInitialized = false;
+
+void initServiceExtensions() {
+  if (_extensionsInitialized)
+    return;
+
+  registerExtension('flutter', _flutter);
+  // TODO: Expose debugDumpApp().
+  // TODO: Expose StatisticsOverlay.
+  registerExtension('flutter.debugPaint', _debugPaint);
+  registerExtension('flutter.timeDilation', _timeDilation);
+
+  _extensionsInitialized = true;
+
+  // Emit an info level log message; this tells the debugger that the Flutter
+  // service extensions are registered.
+  log('Flutter initialized', name: 'flutter', level: 800);
```

Comment:
https://api.dartlang.org/1.13.0/dart-developer/log.html

I think it's designed for users to hook a Logger from the `logging` package up to it.